pkg/controller/operators/olm/operator.go

```diff
@@ -827,6 +828,37 @@ func (a *Operator) syncNamespace(obj interface{}) error {
 		return err
 	}
 
+	// Query OG in this namespace
+	groups, err := a.lister.OperatorsV1().OperatorGroupLister().OperatorGroups(namespace.GetName()).List(labels.Everything())
+
+	// Check if there is a stale multiple OG condition and clear it if existed.
+	if len(groups) == 1 {
```

Comment:
why do this in `syncNamespace` instead of `syncOperatorGroup`? The only time this condition changes is when an operatorgroup has been created or deleted, right?

Reply:
Because in `syncNamespace`, I can query all OG in that namespace to determine if there are multiple OG in the same namespace and add the conditions for MultipleOG accordingly. It is a namespace level check so I think that is an appropriate to check for it. `syncOperatorGroup` is to sync individual OG. If I query all OG in the same namespace for every OG in the same namespace, then there will be redundant calls. I know it is not a big deal given we use cache but still I think this way is more effective. Plus, when a OG is updated, the `syncOperatorGroups` will trigger `syncNamespace` so that will take care of the OG is updated as well.